### PR TITLE
drop support for python 3.6, 3.7, 3.8 and bump dependencies to support 3.11

### PR DESCRIPTION
**Pull request by ekneg54**
This bumps dependencies to support python 3.10 and python 3.11.
It drops support for python 3.6, 3.7, 3.8.

This means we support the last 3 python versions. In perspective of code maintenance this is a good approach in my opinion.

You have to install `librdkafka-dev` and `libhyperscan-dev` before pip install

I had to pin the version of `hyperscan` to `0.2.0`, because higher version needs to be compiled on the target system, which won't work.
The version of `confluent-kafka` is flexible pinned below `1.9.0` in result `1.8.2` is installed as before. But we will get bugfixes I guess.

### CODE REVIEW

**Review by ppcad — APPROVED** [review]

**Review by dtrai2 — CHANGES_REQUESTED** [review]
Hi can you also please check if you can get rid of the custom attrs `min_len_validator` (`logprep.util.validator`)? I think we can get rid of the custom implementation (including it's test) and use the official attrs validator instead. Besides that, did you also verify if the Versioneer library still works with python 3.11? The readme states only 3.6 to 3.10. I don't expect a problem there, but who knows.

File: requirements.in [hunk 1]
```diff
@@ -11,12 +12,10 @@ pygrok
 colorama
 python-dateutil
 pytz
-numpy
+numpy==1.22.1
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply requirements.in:15]: no... numpy has no binary package for python 3.11 and the compilation will fail

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -140,17 +141,16 @@ jobs:
...
-          python -m pip install --upgrade pip
+          sudo apt-get update && sudo apt-get -y install libhyperscan-dev librdkafka-dev
+          pip install --upgrade pip wheel
           pip install -r requirements_dev.txt
 
-      - name: Install semgrep if python version > 3.6
+      - name: Install semgrep
         run: |
           pip install semgrep
-        if: matrix.python-version == 3.7 || matrix.python-version == 3.8 || matrix.python-version == 3.9
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply .github/workflows/ci.yml:149]: semgrep does not support `attrs 22.1.0`

```pwsh
PS> pip-compile ./requirements_dev.in > ./requirements_dev.txt
WARNING: using legacy resolver is deprecated and will be removed in future versions. The default resolver will be change to 'backtracking' in 7.0.0 version. Specify --resolver=backtracking to silence this warning.
Using legacy resolver. Consider using backtracking resolver with `--resolver=backtracking`.
Could not find a version that matches attrs==22.1.0,>=17.3.0,>=19.2.0,~=21.3 (from -r ./requirements.txt (line 17))
Tried: 15.0.0, 15.1.0, 15.1.0, 15.2.0, 15.2.0, 16.0.0, 16.0.0, 16.1.0, 16.1.0, 16.2.0, 16.2.0, 16.3.0, 16.3.0, 17.1.0, 17.1.0, 17.2.0, 17.2.0, 17.3.0, 17.3.0, 17.4.0, 17.4.0, 18.1.0, 18.1.0, 18.2.0, 18.2.0, 19.1.0, 19.1.0, 19.2.0, 19.2.0, 19.3.0, 19.3.0, 20.1.0, 20.1.0, 20.2.0, 20.2.0, 20.3.0, 20.3.0, 21.1.0, 21.1.0, 21.2.0, 21.2.0, 21.3.0, 21.3.0, 21.4.0, 21.4.0, 22.1.0, 22.1.0
Skipped pre-versions: 15.0.0a1
There are incompatible versions in the resolved dependencies:
  attrs==22.1.0 (from -r ./requirements.txt (line 17))
  attrs~=21.3 (from semgrep==0.122.0->-r ./requirements_dev.in (line 7))
  attrs>=17.3.0 (from aiohttp==3.8.3->-r ./requirements.txt (line 7))
  attrs>=19.2.0 (from pytest==7.2.0->-r ./requirements_dev.in (line 2))
```

File: requirements.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (dtrai2) [reply requirements.in:15]: ah okay, good to know, thanks.

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekneg54) [reply .github/workflows/ci.yml:149]: yes... I could downgrade, but I think it is not worth it to downgrade the dependencies of the application for the development environment.

[the remark this answers is not in the document]
Comment (dtrai2) [reply .github/workflows/ci.yml:149]: mh okay, annoying. Then let's hope that there will be a semgrep update soon.

**Comment by ekneg54** [comment]
removed our own min_len validator

**Comment by ekneg54** [comment]
ok... other way around... I have to downgrade attrs, because semgrep doesn't play with the bleeding edge version

**Review by dtrai2 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -230,7 +230,10 @@ Details about the rule language and how to write rules for the processors can be
 
 ### Installation
 
-Python should be present on the system, currently supported are the versions 3.6 - 3.9.
+Python should be present on the system, currently supported are the versions 3.9 - 3.11.
+You have to install `hyperscan` and `rdkafka` before installing logprep. On debian based
+system this is performed by `sudo apt-get update && sudo apt-get -y install libhyperscan-dev librdkafka-dev`.
```
Comment (urdit52) [inline README.md:235]:
```suggestion
systems, this is performed by `sudo apt-get update && sudo apt-get -y install libhyperscan-dev librdkafka-dev`.
```

File: doc/source/getting_started.rst [hunk 4]
```diff
@@ -5,7 +5,7 @@ Getting Started
 Installation
 ============
 
-Python should be present on the system, currently supported are the versions 3.6 - 3.9.
+Python should be present on the system, currently supported are the versions 3.9 - 3.11.
```
Comment (urdit52) [inline doc/source/getting_started.rst:8]:
```suggestion
Python should be present on the system. Currently, Python 3.9 - 3.11 are supported.
```

**Review by urdit52 — CHANGES_REQUESTED** [review]
2 minor change requests.

**Review by urdit52 — APPROVED** [review]
Lgtm. thank you.

**Pull request merged**